[Actions] [ServiceNow] Allow to test close incident for serviceNow

js-jankisalvi opened the pull request:
## Summary

Resolves [#175003](https://github.com/elastic/kibana/issues/175003)

This PR allows to test `closeIncident` subAction in ServiceNow connector test tab.

https://github.com/elastic/kibana/assets/117571355/bbe04be3-ec91-4072-8bda-1e26423fe302

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

ersin-erdal commented on x-pack/plugins/stack_connectors/public/connector_types/servicenow_itsm/types.ts:
```diff
@@ -7,6 +7,11 @@
 
+export enum EventAction {
+  Trigger = 'trigger',
+  Resolve = 'resolve',
```
Very trivial nit: looks like the convention is using all capital in enums.

js-jankisalvi replied:
Updated in https://github.com/elastic/kibana/pull/177026/commits/c541c06fd197336bfc44ed0ee1d084b4cc39822f

ersin-erdal reviewed (approved):
LGTM

cnasikas reviewed (commented):
Hey! I tested and I found the following bugs:

- If I do not select any action and I press run I see the following error:

<img width="857" alt="Screenshot 2024-02-19 at 3 05 42 PM" src="https://github.com/elastic/kibana/assets/7871006/a23c1b6c-4775-4d95-8d25-a8ff9cc0f5c9">

- If I select the trigger action, fill the form (with correlation ID), press run, and then select the resolve action and press again run I see the following error:

<img width="857" alt="Screenshot 2024-02-19 at 3 36 42 PM" src="https://github.com/elastic/kibana/assets/7871006/489e697a-a346-4342-867f-18b8bd5ed715">


Also, I noticed that when creating a rule the Correlation ID is optional when the event action is "Resolve". Do you remember why? I think we should make it mandatory as you have it for the "Test" tab because otherwise users will get an error when the rules tries to close the incident in ServiceNow.

js-jankisalvi commented:
> Also, I noticed that when creating a rule the Correlation ID is optional when the event action is "Resolve". Do you remember why? I think we should make it mandatory as you have it for the "Test" tab because otherwise users will get an error when the rules tries to close the incident in ServiceNow.

Hmm.. Don't remember exactly. We accept `externalId (incident number)` in the API but not sure about UI. Will make it mandatory 👍

js-jankisalvi commented:
> * If I do not select any action and I press run I see the following error:

Fixed by setting Trigger as default event action

> * If I select the trigger action, fill the form (with correlation ID), press run, and then select the resolve action and press again run I see the following error:

Fixed by resetting form fields when changing an event action

[d7e1414](https://github.com/elastic/kibana/pull/177026/commits/d7e141411a491c25a40756ec51e872995ff85f5c)

cnasikas commented:
@elasticmachine merge upstream

cnasikas commented on x-pack/plugins/stack_connectors/public/connector_types/servicenow_itsm/servicenow_itsm_params.tsx:
```diff
@@ -208,16 +257,43 @@ const ServiceNowParamsFields: React.FunctionComponent<
+      ) : null}
       <EuiSpacer size="m" />
-      {selectedActionGroupId !== ACTION_GROUP_RECOVERED ? (
+      {(executionMode !== ActionConnectorMode.Test ||
```
Now that we default the `eventAction` do we still need this? It seems that the `Incident` label is shown in all cases.

cnasikas reviewed (approved):
LGTM! Tested it without issues.
